### PR TITLE
[Inference] Fix Inference Generation Config and Sampling

### PR DESCRIPTION
**Pull request by yuanheng-zhao**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [ ] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs
- [ ] I have installed pre-commit: `pip install pre-commit && pre-commit install`

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/inference/config.py [hunk 1]
```diff
@@ -202,11 +202,12 @@ class InferenceConfig(RPC_PARAM):
     ] = 1.2  # the ratio of prefill sequences to decoding sequences, we do prefill step once the actual value exceeds ratio
     pad_input: bool = False
     early_stopping: Optional[bool] = False
-    top_k: Optional[int] = None
-    top_p: Optional[float] = None
+    top_k: Optional[int] = 50
```
Comment (CjhHa1) [inline colossalai/inference/config.py:205]: why use 50?

File: colossalai/inference/sampler.py [hunk 2]
```diff
@@ -86,23 +71,41 @@ def search_tokens(
...
-        for type in ["temperature", "top_k", "top_p"]:
-            if type in config_dict and config_dict[type] is not None:
-                logits = logit_processor(type, logits, config_dict[type])
+
+    # convert GenerationConfig to dict
+    # temporary fix for compatibility with the usage of RPCInferenceEngine
+    if isinstance(generation_config, GenerationConfig):
+        generation_config = generation_config.to_dict()
+
+    if (repetition_penalty := generation_config.get("repetition_penalty", 1.0)) != 1.0:
```
Comment (CjhHa1) [inline colossalai/inference/sampler.py:80]: no need to use `:=` to set value if we do not use this again

File: colossalai/inference/config.py (hunk 1 above)
Comment (yuanheng-zhao) [reply colossalai/inference/config.py:205]: Corresponds to default value of transformers

File: colossalai/inference/sampler.py (hunk 2 above)
Comment (yuanheng-zhao) [reply colossalai/inference/sampler.py:80]: Notice that `repetition_penalty` is used inside the if block
```python
logits = get_logits_processor("repetition_penalty", logits, repetition_penalty, batch_token_ids)
```

**Review by CjhHa1 — APPROVED** [review]
LGTM

**Pull request merged**